src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,169 @@
+			recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_READY);
+			triggerVideoMetric('ready');
+			// only add the events after the player is ready
+			jwPlayerPlaybackTracker({ event_name: 'video_player_ready' });
+			addBaseTrackingEvents(playerInstance);
+
+			if (onReady) {
+				console.debug('Player ready...');
+				onReady(playerInstance);
+			}
+		});
+
+		playerInstance.on(JWEvents.COMPLETE, () => {
+			if (typeof onComplete === 'function') {
+				onComplete();
+			}
+
+			// Incrementing videos watched count
+			videoIndexRef.current += 1;
+
+			// if stopAutoAdvanceOnExitViewport is false then we don't want to stop the auto advance, keep normal behavior
+			if (!stopAutoAdvanceOnExitViewport) {
+				return;
+			}
+		});
+	};
+
+	const isJWPlayerReady = () => {
+		return typeof window.jwplayer === 'function';
+	};
+
+	const jwPlayerLoaded = () => {
+		console.debug('Player loaded!', jwPlayerContainerEmbedId);
+
+		// Set the max_resolution param for related videos
+		if (typeof window?.jwplayer?.defaults?.related?.file === 'string') {
+			window.jwplayer.defaults.related.file = window.jwplayer.defaults.related.file + '&max_resolution=1280';
+		}
+
+		jwPlayerPlaybackTracker({ event_name: 'video_player_load' });
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_READY);
+		triggerVideoMetric('loaded');
+
+		setConfig(config);
+
+		// TODO: refactor this so we get here the container within jwPlayerContainerEmbedId
+		const playerInstance = window.jwplayer();
+		console.debug('playerInstance: ', playerInstance);
+		registerEventHandlers(playerInstance);
+		setPlayer(playerInstance);
+	};
+
+	const initPlayer = () => {
+		console.debug('initPlayer...');
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_START);
+		jwPlayerPlaybackTracker({ event_name: 'video_player_start_load' });
+
+		const onload = () => {
+			console.debug('Strategy rules embed loaded. Waiting for player...');
+
+			const waitForJWPlayer = waitFor({ eventCheck: isJWPlayerReady });
+			waitForJWPlayer().then(jwPlayerLoaded, () => {
+				console.error('JWPlayer not ready', window?.jwplayer);
+			});
+		};
+
+		if (isJWPlayerReady()) {
+			onload();
+		} else {
+			const script = document.createElement('script');
+			script.async = true;
+			script.src = 'https://cdn.jwplayer.com/v2/sites/cGlKNUnj/placements/embed.js';
+			script.onload = onload;
+			document.getElementsByTagName('head')[0].appendChild(script);
+		}
+	};
+
+	return (
+		<div className={className}>
+			<div
+				id={jwPlayerContainerEmbedId}
+				data-jw-placement-id={'21rL5wJF'}
+				data-jw-playlist={config.playlistUrl}
+				data-jw-recommendations_playlist_id={'FOhaD53w'}
+				data-jw-preroll_ad_tag={
+					'https://pubads.g.doubleclick.net/gampad/ads?iu=%2F5441%2Fwka1b.VIDEO%2Ffeatured%2Fdesktop%2Fucp_desktop-fandom-fv-article%2F_project43-life&sz=640x480&gdfp_req=1&output=xml_vast4&unviewed_position_start=1&env=vp&cust_params=src%3Dtest%26pos%3Dfeatured%26post_id%3D-1'
```

Comment:
Just to be sure, is this tag correct?

Reply:
That's the default 30s blue video. I'll replace it with what I get from [the AdEngine part](https://github.com/Wikia/ad-engine/pull/2141/files#diff-0da6adc9579963d840ecce8ccb2b4ca0a0b28efd4704bdcf64ec01804813091fR32) 😊

Reply:
Does this imply that the current PR is not ready for merging? Or are we planning to have a follow-up PR?

Reply:
Yes 😈 Hi hi hi